components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1533,6 +1540,51 @@ public static String mapSignatureAlgorithm(String signatureAlgorithm) throws Ide
         return mapSignatureAlgorithmForJWSAlgorithm(signatureAlgorithm).getName();
     }
 
+    /**
+     * This method maps the encryption algorithm name defined in identity.xml to a respective
+     * nimbus encryption algorithm.
+     *
+     * @param encryptionAlgorithm name of the encryption algorithm
+     * @return mapped JWEAlgorithm
+     * @throws IdentityOAuth2Exception
+     */
+    public static JWEAlgorithm mapEncryptionAlgorithmForJWEAlgorithm(String encryptionAlgorithm)
+            throws IdentityOAuth2Exception {
+
+        // Parse method in JWEAlgorithm is used to get a JWEAlgorithm object from the algorithm name.
+        JWEAlgorithm jweAlgorithm = JWEAlgorithm.parse(encryptionAlgorithm);
+
+        // Parse method returns a new JWEAlgorithm with requirement set to null if unknown algorithm name is passed.
+        if (jweAlgorithm.getRequirement() != null) {
+            return jweAlgorithm;
+        } else {
+            throw new IdentityOAuth2Exception("Unsupported Encryption Algorithm: " + encryptionAlgorithm + ", in identity.xml");
```

Comment:
length seems to be exceeding 120 characters.

Reply:
Fixed in a later update.